control-plane/pkg/reconciler/source/source.go

```diff
@@ -144,15 +147,40 @@ func (r Reconciler) reconcileConsumerGroup(ctx context.Context, ks *sources.Kafk
 			Extensions: ks.Spec.CloudEventOverrides.Extensions,
 		}
 	}
+
 	if kt, ok := ks.Labels[sources.KafkaKeyTypeLabel]; ok && len(kt) > 0 {
 		expectedCg.Spec.Template.Spec.Configs.KeyType = &kt
 	}
 
+	// TODO: make these parameters below configurable and maybe unexposed
+	if ks.Annotations != nil {
+		expectedCg.Annotations = map[string]string{}
+		if ks.Annotations[keda.AutoscalingClassAnnotation] != "" {
+			expectedCg.Annotations[keda.AutoscalingClassAnnotation] = ks.Annotations[keda.AutoscalingClassAnnotation]
+		}
+		if ks.Annotations[keda.AutoscalingMinScaleAnnotation] != "" {
+			expectedCg.Annotations[keda.AutoscalingMinScaleAnnotation] = ks.Annotations[keda.AutoscalingMinScaleAnnotation]
+		}
+		if ks.Annotations[keda.AutoscalingMaxScaleAnnotation] != "" {
+			expectedCg.Annotations[keda.AutoscalingMaxScaleAnnotation] = ks.Annotations[keda.AutoscalingMaxScaleAnnotation]
+		}
+		if ks.Annotations[keda.KedaAutoscalingPollingIntervalAnnotation] != "" {
+			expectedCg.Annotations[keda.KedaAutoscalingPollingIntervalAnnotation] = ks.Annotations[keda.KedaAutoscalingPollingIntervalAnnotation]
+		}
+		if ks.Annotations[keda.KedaAutoscalingCooldownPeriodAnnotation] != "" {
+			expectedCg.Annotations[keda.KedaAutoscalingCooldownPeriodAnnotation] = ks.Annotations[keda.KedaAutoscalingCooldownPeriodAnnotation]
+		}
+		if ks.Annotations[keda.KedaAutoscalingKafkaLagThreshold] != "" {
+			expectedCg.Annotations[keda.KedaAutoscalingKafkaLagThreshold] = ks.Annotations[keda.KedaAutoscalingKafkaLagThreshold]
+		}
```

Comment:
All ifs seem repetitive can we extract an abstract function?

```
func setAnnotation(in map[string]string, key string, out map[string]string) {
  v, ok := in[key]
  if !ok || v == "" {
    return
  }
  out[key] = v
}
```

Reply:
Related question: when annotations aren't set I think `ks.Annotations[keda.KedaAutoscalingKafkaLagThreshold] != ""` panics